CondDB frontierKey update

smorovic opened the pull request:
#### PR description:

frontierKey in OnlineDBOutputService and CondDBESSource is set to untracked string (this is a parameter relevant only at runtime).

Additional Python changes are included to add a VarParsing parameter and apply the comand line parameter to the OnlineDBOutputService.

#### PR validation:

Tested validity of Python changes with pylint.

smorovic committed:
runUniqueKey VarParsing parameter added for the DQM input source

smorovic committed:
runUniqueKey propagated to beamspot condition-data DQM injector

smorovic committed:
frontierKey changed to untracked parameter. Output module gets the same
parameter instead of checking the file

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-33643/22518

- This PR adds an extra 36KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  DQM/Integration/python/clients/beam_dqm_sourceclient-live_cfg.py  modified in PR(s): #33428
  - File  DQM/Integration/python/clients/beamfake_dqm_sourceclient-live_cfg.py  modified in PR(s): #33428

cmsbuild commented:
A new Pull Request was created by  @smorovic (Srecko Morovic) for master.

It involves the following packages:

CondCore/DBOutputService
CondCore/ESSources
DQM/Integration


@malbouis, @andrius-k, @yuanchao, @kmaeshima, @ErnestaP, @ahmad3213, @cmsbuild, @jfernan2, @tlampen, @ggovi, @pohsun, @rvenditti, @francescobrivio can you please review it and eventually sign? Thanks.
@mmusich, @threus, @batinkov, @tocheng, @battibass this is something you requested to watch as well.
@silviodonato, @dpiparo, @qliphy you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

smorovic commented:
@cmsbuild please test

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-566eb7/14904/summary.html
**COMMIT**: f5558250f229faf4c4044e54c7a60c3c099e068f
**CMSSW**: CMSSW_12_0_X_2021-05-05-2300/slc7_amd64_gcc900
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/33643/14904/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



The workflows 140.53 have different files in step1_dasquery.log than the ones found in the baseline. You may want to check and retrigger the tests if necessary. You can check it in the "files" directory in the results of the comparisons

**Summary**:
 - No significant changes to the logs found
 - Reco comparison results: 1266 differences found in the comparisons
 - DQMHistoTests: Total files compared: 37
 - DQMHistoTests: Total histograms compared: 2662646
 - DQMHistoTests: Total failures: 3678
 - DQMHistoTests: Total nulls: 20
 - DQMHistoTests: Total successes: 2658926
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: -45.699 KiB( 36 files compared)
 - DQMHistoSizes: changed ( 140.53 ): -44.531 KiB      Hcal/DigiRunHarvesting
 - DQMHistoSizes: changed ( 140.53 ): -1.172 KiB       RPC/DCSInfo
 - DQMHistoSizes: changed ( 312.0 ): 0.004 KiB        MessageLogger/Warnings
 - Checked 155 log files, 37 edm output root files, 37 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_0_X_2021-05-05-2300+566eb7/42649/triggerResults): no differences found

smorovic committed:
frontier key added to DQM other input source type cfi files. Will have
no effect when not used from hltd

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-33643/22653

- This PR adds an extra 44KB to repository

cmsbuild commented:
Pull request #33643 was updated. @malbouis, @andrius-k, @yuanchao, @kmaeshima, @ErnestaP, @ahmad3213, @cmsbuild, @jfernan2, @tlampen, @ggovi, @pohsun, @rvenditti, @francescobrivio can you please check and sign again.

jfernan2 commented:
please test

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-566eb7/15071/summary.html
**COMMIT**: e8953732494ed308aa2b5756f79cccbaf5b1ec4f
**CMSSW**: CMSSW_12_0_X_2021-05-12-2300/slc7_amd64_gcc900
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/33643/15071/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



The workflows 140.53 have different files in step1_dasquery.log than the ones found in the baseline. You may want to check and retrigger the tests if necessary. You can check it in the "files" directory in the results of the comparisons

**Summary**:
 - No significant changes to the logs found
 - Reco comparison results: 1260 differences found in the comparisons
 - DQMHistoTests: Total files compared: 37
 - DQMHistoTests: Total histograms compared: 2648242
 - DQMHistoTests: Total failures: 3669
 - DQMHistoTests: Total nulls: 19
 - DQMHistoTests: Total successes: 2644532
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 45.703 KiB( 36 files compared)
 - DQMHistoSizes: changed ( 140.53 ): 44.531 KiB       Hcal/DigiRunHarvesting
 - DQMHistoSizes: changed ( 140.53 ): 1.172 KiB        RPC/DCSInfo
 - Checked 155 log files, 37 edm output root files, 37 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_0_X_2021-05-12-2300+566eb7/42775/triggerResults): no differences found

jfernan2 commented:
+1
11_3_X Backport Tested in Online DQM at P5 with new HLTD rpm

yuanchao commented:
+1
* Modification on python configuration only (built ok)
* tests passed

ggovi commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @silviodonato, @dpiparo, @qliphy (and backports should be raised in the release meeting by the corresponding L2)

silviodonato commented on DQM/Integration/python/clients/beam_dqm_sourceclient-live_cfg.py:
```diff
@@ -385,6 +385,12 @@
+        process.OnlineDBOutputService.frontierKey = cms.untracked.string(options.runUniqueKey)
+        print("Configured frontierKey", options.runUniqueKey)
+    except:
+        pass
```
@smorovic @cms-sw/dqm-l2 @cms-sw/alca-l2 @cms-sw/db-l2 isn't it better to add a message also in the "except" branch ?

smorovic replied:
@silviodonato
with all possible input sources used in this and similar configuration files now having the runUniqueKey VarArgs option, we could also just remove try/except and set frontierKey within process.OnlineDBOutputService definition. Does this seem ok for @cms-sw/dqm-l2 and others?

jfernan2 replied:
Yes, fine from our side. Thanks

smorovic replied:
Pushed the change. If it looks ok, I'll also add it to 11_3_X backport.

smorovic committed:
options.runUniqueKey is added to all input source cfi files, so we can
by default propagate it to all OnlineDBOutputService modules in beamspot
configurations

jfernan2 commented:
please test

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-33643/22699

- This PR adds an extra 20KB to repository

cmsbuild commented:
Pull request #33643 was updated. @malbouis, @andrius-k, @yuanchao, @kmaeshima, @ErnestaP, @ahmad3213, @jfernan2, @tlampen, @ggovi, @pohsun, @rvenditti, @francescobrivio can you please check and sign again.

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-566eb7/15135/summary.html
**COMMIT**: 46a4a074d61a123f86893e9cf32c2f4e62bf582c
**CMSSW**: CMSSW_12_0_X_2021-05-16-2300/slc7_amd64_gcc900
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/33643/15135/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - No significant changes to the logs found
 - Reco comparison results: 8 differences found in the comparisons
 - DQMHistoTests: Total files compared: 37
 - DQMHistoTests: Total histograms compared: 2648242
 - DQMHistoTests: Total failures: 12
 - DQMHistoTests: Total nulls: 1
 - DQMHistoTests: Total successes: 2648207
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: -0.004 KiB( 36 files compared)
 - DQMHistoSizes: changed ( 312.0 ): -0.004 KiB       MessageLogger/Warnings
 - Checked 155 log files, 37 edm output root files, 37 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_0_X_2021-05-16-2300+566eb7/42821/triggerResults): no differences found

jfernan2 commented:
+1

ggovi commented:
+1

yuanchao commented:
+1
* frontierKey in OnlineDBOutputService and CondDBESSource set to untracked string; python module compiled ok
* Add a VarParsing parameter and apply CL parameter to OnlineDBOutputService (technical)
* WF matrix tests and unit tests passed

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @silviodonato, @dpiparo, @qliphy (and backports should be raised in the release meeting by the corresponding L2)

silviodonato commented:
please test
to include the fix https://github.com/cms-sw/cmsdist/pull/6911

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-566eb7/15155/summary.html
**COMMIT**: 46a4a074d61a123f86893e9cf32c2f4e62bf582c
**CMSSW**: CMSSW_12_0_X_2021-05-17-2300/slc7_amd64_gcc900
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/33643/15155/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - No significant changes to the logs found
 - Reco comparison results: 4 differences found in the comparisons
 - DQMHistoTests: Total files compared: 37
 - DQMHistoTests: Total histograms compared: 2648242
 - DQMHistoTests: Total failures: 6
 - DQMHistoTests: Total nulls: 1
 - DQMHistoTests: Total successes: 2648213
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: -0.004 KiB( 36 files compared)
 - DQMHistoSizes: changed ( 312.0 ): -0.004 KiB       MessageLogger/Warnings
 - Checked 155 log files, 37 edm output root files, 37 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_0_X_2021-05-17-2300+566eb7/42835/triggerResults): no differences found

qliphy commented:
+1
